docs/wiki/3_dev_guide/Distribution-Mechanism.md

```diff
@@ -8,32 +8,54 @@ For the tutorial of distributed training, please see [here](../2_user_guide/Dist
-│            │
-│            │ Request
-│            ▼             Init
-│    RAF NCCLCommunicator◄──────RAF MPICommunicator
-│            │                            │
-│ Call       │ Provide Handle             │ Call
-│            ▼                            ▼
-└────►NCCL Collective Op          MPI Library APIs
+    RAF VM / Interpreter            Client Script
+             │                            │
+             │ Dispatch to                │ Set
+             ▼                            ▼
+┌───RAF Op Implementation       RAF VoidCommunicator
+│     (in NCCL dialect)                   │
+│            │                            │ Be
+│            │ Request                    │
+│            ▼             Init           ▼               Be
+│    RAF NCCLCommunicator◄──────RAF Global Communicator◄──────RAF MPICommunicator (default)
+│            │                                                       │
+│ Call       │ Provide Handle                                        │ Call
+│            ▼                                                       ▼
+└────►NCCL Collective Op                                      MPI Library APIs
 
 
                  Architecture Diagram
 ```
 
+Distributed RAF mainly adopts the Single Program Multiple Data (SPMD) programming model driven by MPI-like libraries such as NVIDIA Collective Communications Library (NCCL), which provides a set of collective communication operators and bridges multiple discrete GPUs on numerous servers connected via conventional Ethernet or high-performance Infiniband. By default, RAF uses MPI to launch the Python script. In such a situation, RAF utilizes both MPI and NCCL in a mixed way, where MPI is used to launch the distributed RAF (create many MPI processes) on user-specified nodes and exchange data for initializing NCCL, while NCCL is responsible for transferring and computing tensors among multiple GPUs efficiently.
+
+RAF also offers the ability of using other launchers instead of MPI. RAF provides a set of API to set/get the global communicator. By default, it will be a `MPICommunicator`. Users can switch to use `VoidCommunicator`, so that users can change those distributed infomation freely and use other launchers (e.g. DeepSpeed, torchrun, user-owned multi-process launcher, etc.). When using `VoidCommunicator`, the data exchange for initializing NCCL will be done by using shared file(s), so environment variable `RAF_FILE_STORE_PATH` should be set to a directory.
+
+*P.S.: `VoidCommunicator` now supports single machine only.*
+
+For example, users can run RAF scripts with `RAF_FILE_STORE_PATH=$(pwd) torchrun --standalone --nnodes=1 --nproc_per_node=4 client_script.py`.
+
+```python
+# client_script.py
+from raf import distributed as dist
+
+dist.set_default_communicator("void")
+comm = dist.get_communicator()
+comm.size = 4  # num of GPUs
+comm.rank = os.environ.get("RANK")
+comm.local_rank = 4  # num of GPUs
+comm.local_size = os.environ.get("RANK")
```

Comment:
Will `torchrun` set the RANK? If so, it would be better to mention it; otherwise it might be better to have RANK be explicitly set in the command line (L41).

Reply:
yes, torchrun/deepspeed set the rank